QGIS 3.42.0 fails to build on aarch64 (openSUSE Tumbleweed)

### What is the bug or the crash?

The openSUSE Tumbleweed aarch64 build for QGIS fails with the following error:
26375-[ 1922s] /home/abuild/rpmbuild/BUILD/qgis-3.42.0-build/qgis-3.42.0/src/3d/qgs3dutils.cpp: In static member function ‘static int Qgs3DUtils::openGlMaxClipPlanes(QSurface*)’:
26376:[ 1922s] /home/abuild/rpmbuild/BUILD/qgis-3.42.0-build/qgis-3.42.0/src/3d/qgs3dutils.cpp:982: error: ‘GL_MAX_CLIP_PLANES’ was not declared in this scope
26377-[ 1922s]   982 |       funcs->glGetIntegerv( GL_MAX_CLIP_PLANES, &numPlanes );

(see [here](https://build.opensuse.org/build/home:tenno:branches:Application:Geo/openSUSE_Tumbleweed/aarch64/qgis/_log) for the full build log)

gs3DUtils::openGlMaxClipPlanes() was introduced with  commit dfce7dc2315c4fa2e44cc274f50a220e9b44011a

### Steps to reproduce the issue

1. Go to the [openSUSE OpenBuildService QGIS Project](https://build.opensuse.org/package/show/home:tenno:branches:Application:Geo/qgis) (this being my branched one, that disables the patch silencing the issue)
2. Branch the project or otherwise obtain the files contained therein
3. Start the build on an openSUSE Tumbleweed aarch64/armv7l host with qgis.spec / the commands from qgis.spec
4. Wait until the build fails

### Versions

3.42.0

### Supported QGIS version

- [x] I'm running a supported QGIS version according to [the roadmap](https://www.qgis.org/en/site/getinvolved/development/roadmap.html#release-schedule).

### New profile

- [ ] I tried with a new [QGIS profile](https://docs.qgis.org/latest/en/docs/user_manual/introduction/qgis_configuration.html#working-with-user-profiles)

### Additional context

The disabled patch consists of #ifdefing GL_MAX_CLIP_PLANES
diff --git i/src/3d/qgs3dutils.cpp w/src/3d/qgs3dutils.cpp
index 053c9458611..78ab57b9915 100644
--- i/src/3d/qgs3dutils.cpp
+++ w/src/3d/qgs3dutils.cpp
@@ -972,6 +972,7 @@ int Qgs3DUtils::openGlMaxClipPlanes( QSurface *surface )
 {
   int numPlanes = 6;
 
+#ifdef GL_MAX_CLIP_PLANES
   QOpenGLContext context;
   context.setFormat( QSurfaceFormat::defaultFormat() );
   if ( context.create() )
@@ -982,6 +983,7 @@ int Qgs3DUtils::openGlMaxClipPlanes( QSurface *surface )
       funcs->glGetIntegerv( GL_MAX_CLIP_PLANES, &numPlanes );
     }
   }
+#endif
 
   return numPlanes;
 }

--- a/src/3d/qgs3dutils.cpp
+++ b/src/3d/qgs3dutils.cpp
@@ -49,6 +49,9 @@
 #include <QOpenGLFunctions>
 #include <Qt3DLogic/QFrameAction>
 
+#if !defined(Q_OS_MAC)
+#include <GL/gl.h>
+#endif
 
 #if QT_VERSION < QT_VERSION_CHECK( 6, 0, 0 )
 #include <Qt3DRender/QBuffer>

--- a/src/3d/qgs3dutils.cpp
+++ b/src/3d/qgs3dutils.cpp
@@ -49,7 +49,7 @@
 #include <QOpenGLFunctions>
 #include <Qt3DLogic/QFrameAction>
 
-#if !defined(Q_OS_MAC)
+#if !defined( Q_OS_MAC )
 #include <GL/gl.h>
 #endif
 

